File: Metode-Avansate-de-Programare/lab2/decorator/DelayTaskRunner.java
package sem1_2.decorator;

public class DelayTaskRunner extends AbstractTaskRunner {
    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        try {
            Thread.sleep(3000);
            System.out.println("Am intarziat");
            super.executeOneTask();
        }
        catch (InterruptedException e) {
            super.executeOneTask();
        }
    }
}

File: Metode-Avansate-de-Programare/lab11/lab11/DelayTaskRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab11;

using System;
using System.Threading;

public class DelayTaskRunner : AbstractTaskRunner
{
    public DelayTaskRunner(TaskRunner taskRunner) : base(taskRunner)
    {
    }

    public override void executeOneTask()
    {
        try
        {
            base.executeOneTask();
            Thread.Sleep(1000);
        }
        catch (ThreadInterruptedException e)
        {
            Console.WriteLine(e);
        }
    }
}